### ISSUE TITLE
Refactor MessageBus

### ISSUE THREAD

**Issue by bd339** [agustincordes/Dagonpp#10]
Modules can post messages every frame. At the end of a frame the `MessageBus` will update and create concurrent tasks for the `Executor`. The problem with this: we first construct a `Message` object, then we potentially create a task object sometime later and destroy the `Message` object. We should refactor this, such that message creation and task creation are more tightly coupled as this allows us to remove the "update the MessageBus" step.
In other words, posting a message should immediately schedule a `Callable` for concurrent handling of the message. I.e. schedule a `Callable` that will send the message to all modules, akin to what `MessageBus::update` is doing at the moment.
